src/resolving.rs

```diff
@@ -48,3 +54,113 @@ pub async fn get_custom_resolver(domains: &Collection<Document>, domain: &str) -
+
+    let pipeline = [
+        doc! {
+            "$match": doc! {
+                "_cursor.to": Bson::Null
+            }
+        },
+        doc! {
+            "$lookup": doc! {
+                "from": "domains",
+                "let": doc! {
+                    "local_resolver_contract": "$resolver_contract"
+                },
+                "pipeline": [
+                    doc! {
+                        "$match": doc! {
+                            "$expr": doc! {
+                                "$eq": [
+                                    "$resolver",
+                                    "$$local_resolver_contract"
+                                ]
+                            },
+                            "_cursor.to": Bson::Null
+                        }
+                    }
+                ],
+                "as": "domainData"
+            }
+        },
+        doc! {
+            "$unwind": doc! {
+                "path": "$domainData",
+                "preserveNullAndEmptyArrays": true
+            }
+        },
+        doc! {
+            "$project": doc! {
+                "_id": 0,
+                "resolver_contract": "$resolver_contract",
+                "uri": "$uri",
+                "domain": "$domainData.domain",
+            }
+        },
+    ];
+    let aggregate_options = AggregateOptions::default();
+    let cursor = offchain_resolvers
+        .aggregate(pipeline, aggregate_options)
+        .await;
+    match cursor {
+        Ok(mut cursor) => {
+            while let Some(doc) = cursor.next().await {
+                if let Ok(doc) = doc {
+                    let domain = doc.get_str("domain").unwrap_or_default();
+                    if domain.is_empty() {
+                        continue;
+                    }
+                    // values in config file override onchain events
+                    match (&state.conf).offchain_resolvers.get(domain) {
+                        Some(_) => continue,
+                        None => {
+                            let resolver = OffchainResolver {
+                                resolver_address: doc
+                                    .get_str("resolver_contract")
+                                    .unwrap_or_default()
+                                    .to_owned(),
+                                uri: vec![clean_string(doc.get_str("uri").unwrap_or_default())],
+                            };
+                            state
+                                .dynamic_offchain_resolvers
+                                .lock()
+                                .unwrap()
+                                .insert(domain.to_owned(), resolver);
+                        }
+                    }
+                }
+            }
+        }
+        Err(err) => {
+            println!("Error while building offchain_resolver hashmap from collection offchain_resolvers: {}", err);
+        }
+    }
+}
+
+pub fn is_offchain_resolver(
```

Comment:
I think get_offchain_resolver would make more sense because you get an Option and not a boolean